test/server/http/admin_test.cc

```diff
@@ -23,12 +24,89 @@
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
+using testing::InSequence;
+using testing::Invoke;
 using testing::NiceMock;
+using testing::Ref;
 using testing::_;
 
 namespace Envoy {
 namespace Server {
 
+/**
+ * This is a heap test allocator that works similar to how the shared memory allocator works in
+ * terms of reference counting, etc.
+ */
+class TestAllocator : public Stats::RawStatDataAllocator {
```

Comment:
@jmarantz added a unit test that verifies the json format. PTAL. I had to copy this TestAllocator class as is from thread_local_store_test.cc as I found that it is the easiest way to generate histograms with stores. I do not know if there is any way to make this class across both the files, with out changing much..

Reply:
Can you move the TestAllocator class to test/test_common/utility.h so you don't have duplicated code?

Reply:
Yes. Moved